data/RecordAction.js

```diff
@@ -71,14 +85,56 @@ export class RecordAction {
         this.hidden = hidden;
         this.prepareFn = prepareFn;
         this.recordsRequired = recordsRequired;
+        this.confirm = confirm;
+    }
+
+    meetsRecordRequirement(count) {
+        const required = this.recordsRequired;
+        return isNil(required) ||
+            (isBoolean(required) && (!required || required && count > 0)) ||
+            (isNumber(required) && count === required);
+    }
+
+    async executeAsync({record, selModel, metadata}) {
+        const {confirm} = this;
+        let promise;
+        if (confirm) {
+            const {text, intent} = this;
+            const cfg = {
+                title: !isEmpty(text) ? `${text}?` : 'Proceed?',
+                confirmText: !isEmpty(text) ? text : 'Proceed',
+                confirmIntent: intent
+            };
+
+            if (isObject(confirm)) {
+                Object.assign(cfg, confirm);
+            } else if (isString(confirm)) {
+                cfg.message = confirm;
+            } else if (!isEmpty(text)) {
+                cfg.message = `Are you sure you want to ${text.toLowerCase()} the record?`;
+            } else {
+                cfg.message = 'Are you sure you want to proceed?';
+            }
+
+            promise = XH.confirm(cfg);
+        } else {
+            promise = resolve(true);
+        }
+
+        const confirmed = await promise;
+        if (!confirmed) return;
+
+        this.actionFn({action: this, record, selModel, metadata});
     }
 }
 
 /**
  * @callback ActionCb - called when the action's UI element is clicked or otherwise triggered.
```

Comment:
would now be the time to consider simplifying this by just having a single bundle of records rather than 'record' and 'selection'.  Do we ever need to know what the primary record clicked on is?  If we did, could that be buried in the 'p.metadata'.?   Would any action operate on both the primary record and the selection?  Mostly thinking outloud here